FogCreek/Glitch-Community
# Coconut restaurant: anon users shouldn't see an add to collections button

## Links
* remix: https://coconut-restaurant.glitch.me
* case: https://glitch.manuscript.com/f/cases/3328445/
* this bug fix got me thinking about popovers more and I documented some of my thoughts in a comment in the pr but also on manuscript: https://glitch.manuscript.com/f/cases/3328381/Build-Popover-component (feel free to tell me if it's not clear)

## GIF/Screenshots:
Before: 
<img width="535" alt="Screen Shot 2019-05-02 at 2 25 17 PM" src="https://user-images.githubusercontent.com/6620164/57097624-275a0680-6ce6-11e9-9b61-12c8200f3001.png">
After:
<img width="562" alt="Screen Shot 2019-05-02 at 2 24 42 PM" src="https://user-images.githubusercontent.com/6620164/57097581-101b1900-6ce6-11e9-9e98-d78ba7a79c3f.png">

## Changes:
* anonymous users will only be able to delete they are admin to or leave projects they are apart of otherwise they never see the dropdowns. anon users will no longer see a broken add to collections button, nor will they be able to feature/unfeature/pin/unpin their projects (until they sign up for an account)

## How To Test:
* logout and create a new project 
* grab your token from local storage (the editor is always on glitch.com so there's no access to this token in the remix)
* go to the remix and add use the token to login
* check out your profile page and click on the dropdown on your newly minted project notice you can only delete
* if you ever join a project I believe you will also see the leaveProject option there (I haven't confirmed this but I trust the system) 
* check out the user/team/collection/search pages and notice you do not see dropdowns to add a project to a collection anymore
* verify stuff still looks good when you are logged in.

## Feedback I'm looking for:
Does this code feel understandable and am I missing any use cases? I don't really think this is a long term solution but a giant refactor felt like something that could happen later. I tried to document in the popover component case, my thoughts on how we could consider approaching this down the road.
